src/RawRabbit/Common/NamingConventions.cs

```diff
@@ -64,15 +66,40 @@ private string GetSubscriberQueueSuffix(Type messageType)
 			return sb.ToString();
 		}
 
-		private string GetApplicationName()
+		private static string GetApplicationName()
 		{
-			return Directory
-				.GetCurrentDirectory()
-				.Split('\\')
-				.Select(d => d.ToLower())
-				.Last(d => !_disallowedDirectoryNames.Contains(d))
-				.Split('.')
-				.Last();
+			return GetApplicationNameFromCommandLine(Environment.CommandLine);
```

Comment:
Why are we calling `GetApplicationNameFromCommandLine` here? IMHO `GetApplicationName` should contain the logic and be called with `Environment.CommandLine`